src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -158,15 +158,15 @@ public RunResult run(final ProductRunData productRunData) throws DetectUserFrien
         return runResult;
     }
 
-    private AggregateOptions determineAggregationStrategy(final String aggregateName, final UniversalToolsResult universalToolsResult) {
+    private AggregateOptions determineAggregationStrategy(final String aggregateName, final boolean aggregateDirect, final UniversalToolsResult universalToolsResult) {
         if (StringUtils.isNotBlank(aggregateName)) {
             if (universalToolsResult.anyFailed()) {
-                return AggregateOptions.aggregateButSkipEmpty(aggregateName);
+                return AggregateOptions.aggregateButSkipEmpty(aggregateName, aggregateDirect);
             } else {
-                return AggregateOptions.aggregateAndAlwaysUpload(aggregateName);
+                return AggregateOptions.aggregateAndAlwaysUpload(aggregateName, aggregateDirect);
             }
         } else {
-            return AggregateOptions.doNotAggregate();
+            return AggregateOptions.doNotAggregate(aggregateDirect);
```

Comment:
If we are not aggregating, why would we care whether or not the aggregate is direct?